src/server/ApiRoutes/Hold.js

```diff
@@ -69,6 +72,31 @@ function postHoldAPI(
 }
 
 /**
+ * mapLocationDetails(locations)
+ * The function extracts the details of the delivery locations from the location.js and
+ * locationCodes.js based on the locastion id we get from deliveryLocationsByBarcode API.
+ *
+ * @param {array} locations
+ * @return {array}
+ */
+function mapLocationDetails(locations) {
+  locations.map(loc => {
+    _mapObject(locationCodes, (c) => {
```

Comment:
The problem with this is that it's going to loop through every single location code each time for however many locations there are. There are 156 location codes so if there are 5 delivery locations, it will do about 156 * 5! (until it finds the code).

I was hoping you could pass in the `location`'s id into `LocationsCodes` but that won't match up because the codes are slightly different. We should talk to Kate and Kevin to see if the location codes mapping is correct.